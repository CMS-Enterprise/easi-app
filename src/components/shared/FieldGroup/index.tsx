import React, { ReactNode, ReactNodeArray } from 'react';
import classnames from 'classnames';

type FieldGroupProps = {
  children: ReactNode | ReactNodeArray;
  className?: string;
  error?: boolean;
  scrollElement?: string;
};
const FieldGroup = ({
  children,
  className,
  error,
  scrollElement
}: FieldGroupProps) => {
  const fieldGroupClasses = classnames(
    'usa-form-group',
    { 'usa-form-group--error': error },
    className
  );
  return (
    <div className={fieldGroupClasses} data-scroll={scrollElement}>
      {children}
    </div>
  );
};

export default FieldGroup;
