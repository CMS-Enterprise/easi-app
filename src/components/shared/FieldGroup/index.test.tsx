import React from 'react';
import { render, screen } from '@testing-library/react';

import FieldGroup from './index';

describe('The FieldGroup component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<FieldGroup>Test</FieldGroup>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct classes', () => {
    render(
      <FieldGroup error data-testid="field-group">
        <div />
      </FieldGroup>
    );
    expect(screen.getByTestId('field-group')).toHaveClass(
      'usa-form-group--error'
    );
  });
});
