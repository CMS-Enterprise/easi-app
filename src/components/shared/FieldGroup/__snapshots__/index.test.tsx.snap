// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The FieldGroup component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="usa-form-group"
  >
    Test
  </div>
</DocumentFragment>
`;
